build/detect-upstream.sh

```diff
@@ -85,9 +87,37 @@ parse_make(){
   fi
 }
 
+parse_manifests(){
+  echo "Scanning manifests..."
+
+  quay_ocm='quay\.io\/open-cluster-management'
+
+  manifests=("$@")
+  upstream_count=0
+  for filename in "${manifests[@]}"; do
+    upstream_count_curr=0
+    while read -r line; do
+      if [[ ${line} =~ $quay_ocm ]]; then
+        upstream_count_curr=$((upstream_count_curr + 1))
+      fi
+    done < "$filename"
+    if [[ $upstream_count_curr > 0 ]]; then
+      echo "${filename}: $upstream_count_curr Stolostron Quay URLs found"
+    fi
+    upstream_count=$((upstream_count_curr + upstream_count))
+  done
```

Comment:
sure, I can do it like that instead. I wrote it like this because I thought we only wanted to parse the Makefile and the manifests for each repo. If we want to detect all quay urls over the entire repo, grep works a lot better.

Reply:
Sounds good--I go back and forth in my head, so we should probably keep on rolling with it. 🙂